src/include/miopen/batchnorm/problem_description.hpp

```diff
@@ -267,6 +302,43 @@ struct MIOPEN_INTERNALS_EXPORT ProblemDescription : ProblemDescriptionBase, Prob
     std::string ComputeInLayout() const { return ComputeLayout(xDesc); }
     std::string ComputeOutLayout() const { return ComputeLayout(yOrDyDesc); }
     std::string ComputeDinLayout() const { return ComputeLayout(dxDesc); }
+
+    size_t GetSpatialDims() const { return spatial_dim; }
+
+    std::size_t GetBatchSize() const { return GetN5(GetSpatialDims(), xDesc.GetLengths()); }
+    std::size_t GetChannel() const { return GetC5(GetSpatialDims(), xDesc.GetLengths()); }
+    std::size_t GetHeight() const { return GetH5(GetSpatialDims(), xDesc.GetLengths()); }
+    std::size_t GetWidth() const { return GetW5(GetSpatialDims(), xDesc.GetLengths()); }
+    std::size_t GetDepth() const { return GetD5(GetSpatialDims(), xDesc.GetLengths()); }
+
+    std::string GetDirectionStr() const
+    {
+        std::string s;
+
+        switch(direction)
+        {
+        case Direction::ForwardInference: s = "F2b0"; break;
+        case Direction::ForwardTraining: s = "F1b0"; break;
+        case Direction::Backward: s = "F0b1"; break;
+        default: assert(false);
```

Comment:
I'd rather `MIOPEN_THROW` here, because it would be an error that should fail the CI no matter what